lib/models/model.dart

```diff
@@ -0,0 +1,18 @@
+import 'dart:math';
+import 'dart:convert';
+
+String randomId(int len) {
+  Random random = Random.secure();
+  List<int> values = List<int>.generate(len, (int i) => random.nextInt(255));
+  return base64UrlEncode(values);
+}
+
+abstract class Model {
+  String id;
+
+  Model({String? id}) : id = id ?? randomId(15);
+
+  Model.fromJson(Map<String, dynamic> json) : id = json['id'];
```

Comment:
I write it here but it's the same everywhere, I have no problem that you write your ``fromJson`` and ``toJson`` by hand but there is a package called ``json_serializable`` to make your life easier x)
https://pub.dev/packages/json_serializable